### PR TITLE
Might be small but it's something :)

### PR DESCRIPTION
**Pull request by drnoah**
nothing special pull created to just give you notice ;) This is ment to be not merged ;)

### CODE REVIEW

File: src/main/java/org/rxbooter/injector/core/Node.java [hunk 1]
```diff
@@ -22,7 +22,7 @@
 
 import org.rxbooter.injector.Injector;
 import org.rxbooter.injector.Key;
-
+//guys seem to be in love with lombock so maybe you wanna do it ;P
 /**
```
Comment (siy) [inline src/main/java/org/rxbooter/injector/core/Node.java:26]: I'm trying to keep dependencies as minimal as possible. Lombok is not the one which is strictly necessary here.

File: src/main/java/org/rxbooter/injector/core/supplier/SupplierFactory.java [hunk 2]
```diff
@@ -140,6 +140,8 @@ private SupplierFactory() {}
 
     @SuppressWarnings("unchecked")
     private static <T> ThrowingSupplier<T> mapMethodParameters(Method method, Supplier<?>[] suppliers) {
+
+       //that is way to fugly but probably no way to not have it
         switch (method.getParameterCount()) {
```
Comment (siy) [inline src/main/java/org/rxbooter/injector/core/supplier/SupplierFactory.java:145]: Definitely. This approach is result of a number of experiments and it seems simplest (and fastest) one.

File: src/main/java/org/rxbooter/injector/core/supplier/SupplierFactory.java [hunk 3]
```diff
@@ -167,7 +169,8 @@ private SupplierFactory() {}
 
     private static <T> ThrowingSupplier<T> mapConstructorParameters(Constructor<T> constructor, Supplier<?>[] suppliers) {
         switch (constructor.getParameterCount()) {
-            case  0: return () -> constructor.newInstance();
+            //that is way to fugly but probably no way to not have it
+            case  0: return constructor::newInstance;
```
Comment (siy) [inline src/main/java/org/rxbooter/injector/core/supplier/SupplierFactory.java:173]: I'd prefer to keep all this code consistent, this is why I didn't use method reference here.

File: src/main/java/org/rxbooter/injector/core/supplier/Suppliers.java [hunk 4]
```diff
@@ -42,13 +42,14 @@ private Suppliers() {
 
         return new Supplier<T>() {
             private final Supplier<T> defaultDelegate = this::init;
-            private final AtomicBoolean marker = new AtomicBoolean();
+            private final AtomicBoolean noWayIKnowAtFirstGlanceWhatThisDoesVariable = new AtomicBoolean();
```
Comment (siy) [inline src/main/java/org/rxbooter/injector/core/supplier/Suppliers.java:45]: This variable is necessary to make sure that piece of code will be executed only once. It also 'marks' state as changed (initialized), this is why variable is called 'marker'.

File: src/test/java/org/rxbooter/injector/core/KeyTest.java [hunk 5]
```diff
@@ -78,6 +78,7 @@ public void keyFollowsEqualsContract() throws Exception {
         assertThat(key1).isEqualTo(key1);
         assertThat(key2).isEqualTo(key2);
 
+        //Key compared to string ??
         assertThat(key1.equals("")).isFalse();
```
Comment (siy) [inline src/test/java/org/rxbooter/injector/core/KeyTest.java:82]: This test verifies that .equals() contract is followed. And .equals() can receive any other object as input, so check is perfectly legal and validates that objects of other type will be considered not equal.

File: src/main/java/org/rxbooter/injector/core/supplier/Suppliers.java (hunk 4 above)
Comment (drnoah) [reply src/main/java/org/rxbooter/injector/core/supplier/Suppliers.java:45]: so maybe name like initialized would be good.

File: src/main/java/org/rxbooter/injector/core/supplier/SupplierFactory.java (hunk 3 above)
Comment (drnoah) [reply src/main/java/org/rxbooter/injector/core/supplier/SupplierFactory.java:173]: yeah expected that :)

File: src/main/java/org/rxbooter/injector/core/Node.java (hunk 1 above)
Comment (drnoah) [reply src/main/java/org/rxbooter/injector/core/Node.java:26]: :)

File: src/test/java/org/rxbooter/injector/core/KeyTest.java (hunk 5 above)
Comment (drnoah) [reply src/test/java/org/rxbooter/injector/core/KeyTest.java:82]: I have put equals and hascode for granted as autogenerated.

Comment (siy) [reply src/test/java/org/rxbooter/injector/core/KeyTest.java:82]: This class has some specifics - boolean supplier is not taken into account for equals and hashCode.

File: src/main/java/org/rxbooter/injector/Injector.java [hunk 6]
```diff
@@ -107,7 +107,7 @@
 
     /**
      * Configure {@link Injector} with specified classes. Each class can provide two variants of configuration data:
-     * class may implement {@link Module} interface (or be a sublass of {@link AbstractModule}) and/or describe
+     * class may implement {@link Module} interface (or be a subclass of {@link AbstractModule}) and/or describe
```
Comment (siy) [inline src/main/java/org/rxbooter/injector/Injector.java:110]: Fix this in next commit to develop.

**Pull request closed without merging**